### ISSUE TITLE
Can Fastcorrect be used for Error Correction in English?

### ISSUE THREAD

**Issue by Amber921463001** [microsoft/NeuralSpeech#27]
If I want to do English error correction, can I use this model? Is it possible to pre-train and fine-tune with English corpus?

**Comment by Amber921463001** [comment]
Is the pre-trained data format the same as the fine-tuned data format? If I want to pretrain with English corpus，in addition to English sentencepiece model  and the English dict ，what else do I need to prepare？Thank you！

**Comment by YichongLeng** [comment]
The pre-trained data format is the same as the finetuned data format. I think corpus, sentencepiece and dictionary is enough.

**Comment by Amber921463001** [comment]
You said you use the internal 400M unpaired text data, I want to sure that whether 400M is 400 millions sentences or 400 MiB dataset？Thank you！

**Comment by Amber921463001** [comment]
By the way,do I have to use English n-gram.txt when running align_cal_werdur_v2.py ?

**Comment by YichongLeng** [comment]
> You said you use the internal 400M unpaired text data, I want to sure that whether 400M is 400 millions sentences or 400 MiB dataset？Thank you！

400M sentences.


> By the way,do I have to use English n-gram.txt when running align_cal_werdur_v2.py ?

No. You can skip this logic, which will not account much if the data is large enough.

**Comment by Amber921463001** [comment]
I see. Thank you for your reply！

**Comment by Amber921463001** [comment]
I saw you used chinese_char_sim.txt and sim_prun_char.txt when running add_noise.py. What is the difference between these two files? I only have one file now , in which the  first column is an English word, and the second column is some sound-like words for that English word.

**Comment by Amber921463001** [comment]
Could you tell me your final loss in your pre-training phase？

**Comment by YichongLeng** [comment]
> I saw you used chinese_char_sim.txt and sim_prun_char.txt when running add_noise.py. What is the difference between these two files? I only have one file now , in which the first column is an English word, and the second column is some sound-like words for that English word.

One file is fine.

> Could you tell me your final loss in your pre-training phase？

The pre-training loss is relevant to the ratio of noise (the noisier, the higher loss) and might be different in different language.


> I have trained the model with English corpus, but in the evaluation phase, the English words that model predicted are linked together without spaces between words. Such like “therecssomsnbcanprypathsthateolecanprakinprsucss” . Could you please tell me how to solve this problem? Maybe I should modify the decoder code? But I don't know which file should be modified

I think this is a strange problem and maybe you can check the decode function in https://github.com/microsoft/NeuralSpeech/blob/master/FastCorrect/eval_aishell.py#L85.

**Comment by g-karthik** [comment]
@Amber921463001 did you figure this out and were able to get it working for English? If so, could you please share your English-working version? I would like to reproduce this for my experiments.

**Comment by g-karthik** [comment]
Also, @YichongLeng, it looks like you were saying you can skip the source-target alignment step entirely.

> No. You can skip this logic, which will not account much if the data is large enough.

Would it be possible for you to share some details on the exact steps to follow / skip for English ASR EC in the existing README for FastCorrect?
